nautobot/dcim/models/power.py

```diff
@@ -137,31 +174,121 @@ class PowerFeed(PrimaryModel, PathEndpoint, CableTermination):
+            ["power_panel", "name"],
+            ["power_panel", "breaker_position"],
+        ]
+        indexes = [
+            models.Index(fields=["power_panel", "breaker_position"]),
+        ]
 
     def __str__(self):
         return self.name
 
     def clean(self):
         super().clean()
 
-        # Rack must belong to same Location as PowerPanel
-        if self.rack and self.rack.location != self.power_panel.location:  # pylint: disable=no-member
-            raise ValidationError(
-                f"Rack {self.rack} ({self.rack.location}) and "  # pylint: disable=no-member
-                f"power panel {self.power_panel} ({self.power_panel.location}) are in different locations"
-            )
+        # Rack must belong to same location hierarchy as PowerPanel
+        if self.rack and self.power_panel.location:  # pylint: disable=no-member
+            if self.rack.location not in self.power_panel.location.ancestors(  # pylint: disable=no-member
+                include_self=True
+            ) and self.power_panel.location not in self.rack.location.ancestors(include_self=True):  # pylint: disable=no-member
+                raise ValidationError(
+                    {
+                        "rack": f'Rack "{self.rack}" ({self.rack.location}) and '  # pylint: disable=no-member
+                        f'power panel "{self.power_panel}" ({self.power_panel.location}) '  # pylint: disable=no-member
+                        f"are not in the same location hierarchy."
+                    }
+                )
 
         # AC voltage cannot be negative
         if self.voltage < 0 and self.supply == PowerFeedSupplyChoices.SUPPLY_AC:
             raise ValidationError({"voltage": "Voltage cannot be negative for AC supply"})
 
+        # Destination panel validation
+        if self.destination_panel:
+            # Cannot feed into the same panel
+            if self.destination_panel == self.power_panel:
+                raise ValidationError({"destination_panel": "A power feed cannot connect a panel to itself"})
+            # TODO: add loop detection when graph structure is implemented for path tracing
+
+        # Enforce mutual exclusivity between cable connections and destination_panel
+        if self.destination_panel and self.cable:
+            raise ValidationError(
+                {
+                    "destination_panel": "Cannot specify a destination panel when the power feed is connected via cable. "
+                    "Power feeds can either connect to a panel OR be cabled to an endpoint, but not both."
+                }
+            )
+
+        if self.cable and self.destination_panel:
+            raise ValidationError(
+                {
+                    "__all__": "This power feed cannot be connected via cable when a destination panel is specified. "
+                    "Power feeds can either connect to a panel OR be cabled to an endpoint, but not both."
+                }
+            )
```

Comment:
One of these two is redundant, isn't it?

```suggestion
        # Enforce mutual exclusivity between cable connections and destination_panel
        if self.destination_panel and self.cable:
            raise ValidationError(
                {
                    "destination_panel": "Cannot specify a destination panel when the power feed is connected via cable. "
                    "Power feeds can either connect to a panel OR be cabled to an endpoint, but not both."
                }
            )
```